index.js

```diff
@@ -298,68 +370,105 @@ function cmpType (a, b) {
   return typeOrder[a.type] - typeOrder[b.type]
 }
 
-DB.prototype._onpt = function (pt, seen, cb) {
+// Given a node by its version, this collects the node itself, and also
+// recursively climbs all ways and relations that the node (or its referers)
+// are referred to by.
+// OsmVersion, { OsmVersion: Boolean } -> [OsmDocument]
+DB.prototype._collectNodeAndReferers = function (version, seenAccum, cb) {
   cb = once(cb || noop)
   var self = this
-  var link = pt.value.toString('hex')
-  if (has(seen, link)) return cb(null, [])
-  seen[link] = true
-  var res = [], added = {}, pending = 2
-  self.log.get(link, function (err, doc) {
-    if (doc && doc.value && doc.value.k && doc.value.v) {
-      addDoc(doc.value.k, link, doc.value.v)
-    }
-    if (--pending === 0) cb(null, res)
+  if (has(seenAccum, version)) return cb(null, [])
+  var res = [], added = {}, pending = 1
```

Comment:
I made the `--pending <= 0` changes, but see my comment below re error handling.